### PR TITLE
DDF-3718 Updated docs unit test and fixed broken anchors

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
Updates the unit tests in the documentation to check for broken anchors (internal links), freemarker stack traces and missing images.

The primary changes take place in DocumentationTest.java and the rest of the changes are to fix broken links found using the new unit test.

#### Who is reviewing it? 
@vinamartin @garrettfreibott @ahoffer @sblackmore 

#### Select relevant component teams: 
@codice/docs 

#### Choose 2 committers to review/merge the PR. 
@ricklarsen @rzwiefel 

#### How should this be tested? (List steps with links to updated documentation)
Inspect doc changes, review test design, run full build

#### What are the relevant tickets?
[DDF-3718](https://codice.atlassian.net/browse/DDF-3718)

#### Checklist:
- [x] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/test/java/ddf/docs/DocumentationTest.java [hunk 1]
```diff
@@ -20,36 +20,93 @@
...
+                            || s.toLowerCase().contains(FREEMARKER_MSG)
+                            || s.contains(BASE64_MISSING));
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }));
+  }
+
+  @Test
+  public void testBrokenAnchorsPresent() throws IOException, URISyntaxException {
```
Comment (ricklarsen) [inline distribution/docs/src/test/java/ddf/docs/DocumentationTest.java]: ❤️

**Comment by vinamartin** [comment]
❓ I am fine with this change and I'm happy that we've fixed all those broken links. Awesome stuff! The only thing I'm a little confused about is how other projects typically deal with this. I found this: http://www.writethedocs.org/guide/tools/testing/#testing-your-documentation . 
I'm fine with the unit test approach, but to me, this feels like something that should break during compile. I'm not entirely sure how that would work, it seems to me like most tools really should have that handled. Is that something that isn't possible with the current way we build docs?

**Comment by mcalcote** [comment]
@ricklarsen what happens when the document can't be rendered during the build? Does the build fail?

If it does, then I would argue we are alright, though those linked tools may help reduce time to diagnosis. 

I think that these tests provide a pretty solid net since we ensure the docs were actually rendered in the first place in the expected location and that they do not contain any of the the known issues they have had in the past (or may have in the future).

I am not against investigating those additional libraries for the benefits they offer. Maybe we should create a spike?

File: distribution/docs/src/main/java/ddf/docs/NoOp.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
```
Comment (mcalcote) [inline distribution/docs/src/main/java/ddf/docs/NoOp.java]: Apparently this class is needed to have surefire run the tests in the docs module.

**Comment by mcalcote** [comment]
After rebasing I found another broken anchor, updated. Build now

**Pull request merged**